### DISCUSSION TITLE
Should we rename the "operator" and the "operatorFee" arguments?

### DISCUSSION THREAD

**Discussion by PaulRBerg** [sablier-labs/evm-monorepo#234] [General]
In #220, I have implemented operator fees (https://github.com/sablierhq/v2-core/issues/208) by adding the two following arguments in the create functions:

- `operator`
- `operatorFee`

But @andreivladbrg has pointed out to me that this term is already used by the [ERC-721](https://eips.ethereum.org/EIPS/eip-721) standard to refer to an approved account that can transfer the NFT on behalf of the owner.

I agree with Andrei that this is not good - it's confusing to have the same term referring to two different concepts in the same code base.

Alternatives:

- `agent`
- `broker`
- `intermediary` (though this doesn't sound nice in the wider context of decentralization)

None is better than `operator`, though.

**Comment by andreivladbrg** [comment]
Given that this model of fees is very similar to a referral program, what do you think about:
- `referralFee`
-  `referre`

**Reply by PaulRBerg** [reply]
Interesting idea! Let me sleep on it.

Though it would certainly not be `referre` - it would be `referrer`, and at any rate we would bundle these parameters in a `Referrer` struct (analogous to [`Broker`](https://github.com/sablierhq/v2-core/issues/237)).

**Reply by PaulRBerg** [reply]
Actually, I still like `broker` slightly more than `referrer`, because it seems to me that the former captures the business flow more accurately. A third-party frontend that integrates Sablier does not "refer" the user to Sablier - the user may already know about Sablier. That frontend is just a tool to access the Sablier protocol - therefore, just like a broker that purchases shares on behalf of a customer, a third-party frontend brokerages on behalf of the web3 user to create Sablier streams for them.

**Reply by PaulRBerg** [reply]
We might need a third opinion from @razgraf and @gavriliumircea to adjudicate this.

**Reply by razgraf** [reply]
Aave's using `referral` and it makes sense. But I do enjoy the sound of `broker` more. No fancy reason, just sounds sexy.

**Reply by PaulRBerg** [reply]
Thanks, Razvan.

Aave's case is a bit different because their rewards are continuous, whereas our fees are one-off (ironically). I think that `referral` fits better in a continuous context.

Separately, I envision Sablier V2 being used by such corporate entities as Coinbase. For these users, "broker" is a much better fitting term.

**Reply by gavriliumircea** [reply]
I think for our case the "broker" would be better. It makes more sense for me to use this one instead of "refferer".

**Reply by PaulRBerg** [reply]
Thanks, Mircea. `broker` is the winner.

Gonna lock the convo.
